lgc/disassembler/PalMetadataRegs.cpp

```diff
@@ -0,0 +1,439 @@
+    {0x2D1B, "SPI_SHADER_USER_DATA_HS_15"},
+    {0x2D1B, "SPI_SHADER_USER_DATA_LS_15"},
+    {0x2D1C, "SPI_SHADER_USER_DATA_HS_16"},
+    {0x2D1C, "SPI_SHADER_USER_DATA_LS_16"},
+    {0x2D1D, "SPI_SHADER_USER_DATA_HS_17"},
+    {0x2D1D, "SPI_SHADER_USER_DATA_LS_17"},
+    {0x2D1E, "SPI_SHADER_USER_DATA_HS_18"},
+    {0x2D1E, "SPI_SHADER_USER_DATA_LS_18"},
+    {0x2D1F, "SPI_SHADER_USER_DATA_HS_19"},
+    {0x2D1F, "SPI_SHADER_USER_DATA_LS_19"},
+    {0x2D20, "SPI_SHADER_USER_DATA_HS_20"},
+    {0x2D20, "SPI_SHADER_USER_DATA_LS_20"},
+    {0x2D21, "SPI_SHADER_USER_DATA_HS_21"},
+    {0x2D21, "SPI_SHADER_USER_DATA_LS_21"},
+    {0x2D22, "SPI_SHADER_USER_DATA_HS_22"},
+    {0x2D22, "SPI_SHADER_USER_DATA_LS_22"},
+    {0x2D23, "SPI_SHADER_USER_DATA_HS_23"},
+    {0x2D23, "SPI_SHADER_USER_DATA_LS_23"},
+    {0x2D24, "SPI_SHADER_USER_DATA_HS_24"},
+    {0x2D24, "SPI_SHADER_USER_DATA_LS_24"},
+    {0x2D25, "SPI_SHADER_USER_DATA_HS_25"},
+    {0x2D25, "SPI_SHADER_USER_DATA_LS_25"},
+    {0x2D26, "SPI_SHADER_USER_DATA_HS_26"},
+    {0x2D26, "SPI_SHADER_USER_DATA_LS_26"},
+    {0x2D27, "SPI_SHADER_USER_DATA_HS_27"},
+    {0x2D27, "SPI_SHADER_USER_DATA_LS_27"},
+    {0x2D28, "SPI_SHADER_USER_DATA_HS_28"},
+    {0x2D28, "SPI_SHADER_USER_DATA_LS_28"},
+    {0x2D29, "SPI_SHADER_USER_DATA_HS_29"},
+    {0x2D29, "SPI_SHADER_USER_DATA_LS_29"},
+    {0x2D2A, "SPI_SHADER_USER_DATA_HS_30"},
+    {0x2D2A, "SPI_SHADER_USER_DATA_LS_30"},
+    {0x2D2B, "SPI_SHADER_USER_DATA_HS_31"},
+    {0x2D2B, "SPI_SHADER_USER_DATA_LS_31"},
+    {0x2D32, "SPI_SHADER_USER_ACCUM_LSHS_0"},
+    {0x2D33, "SPI_SHADER_USER_ACCUM_LSHS_1"},
+    {0x2D34, "SPI_SHADER_USER_ACCUM_LSHS_2"},
+    {0x2D35, "SPI_SHADER_USER_ACCUM_LSHS_3"},
+    {0x2D3D, "SPI_SHADER_PGM_RSRC2_LS_HS"},
+    {0x2D4A, "SPI_SHADER_PGM_RSRC1_LS"},
+    {0x2D4B, "SPI_SHADER_PGM_RSRC2_LS"},
+    {0x2D4C, "SPI_SHADER_USER_DATA_LS_0"},
+    {0x2D4D, "SPI_SHADER_USER_DATA_LS_1"},
+    {0x2D4E, "SPI_SHADER_USER_DATA_LS_2"},
+    {0x2D4F, "SPI_SHADER_USER_DATA_LS_3"},
+    {0x2D50, "SPI_SHADER_USER_DATA_LS_4"},
+    {0x2D51, "SPI_SHADER_USER_DATA_LS_5"},
+    {0x2D52, "SPI_SHADER_USER_DATA_LS_6"},
+    {0x2D53, "SPI_SHADER_USER_DATA_LS_7"},
+    {0x2D54, "SPI_SHADER_USER_DATA_LS_8"},
+    {0x2D55, "SPI_SHADER_USER_DATA_LS_9"},
+    {0x2D56, "SPI_SHADER_USER_DATA_LS_10"},
+    {0x2D57, "SPI_SHADER_USER_DATA_LS_11"},
+    {0x2D58, "SPI_SHADER_USER_DATA_LS_12"},
+    {0x2D59, "SPI_SHADER_USER_DATA_LS_13"},
+    {0x2D5A, "SPI_SHADER_USER_DATA_LS_14"},
+    {0x2D5B, "SPI_SHADER_USER_DATA_LS_15"},
+    {0x2E00, "COMPUTE_DISPATCH_INITIATOR"},
+    {0x2E07, "COMPUTE_NUM_THREAD_X"},
+    {0x2E08, "COMPUTE_NUM_THREAD_Y"},
+    {0x2E09, "COMPUTE_NUM_THREAD_Z"},
+    {0x2E12, "COMPUTE_PGM_RSRC1"},
+    {0x2E13, "COMPUTE_PGM_RSRC2"},
+    {0x2E15, "COMPUTE_RESOURCE_LIMITS"},
+    {0x2E18, "COMPUTE_TMPRING_SIZE"},
+    {0x2E24, "COMPUTE_USER_ACCUM_0"},
+    {0x2E25, "COMPUTE_SHADER_CHKSUM"},
+    {0x2E25, "COMPUTE_USER_ACCUM_1"},
```

Comment:
Same as on the previous LLVM review [D95619](https://reviews.llvm.org/D95619), there are register numbers that have different names or different meaning between architectures.
If you don’t want to fix that here, but maybe add a `FIXME`?

Reply:
Good point. Like you and Jay said over there, this is only to make it easier to read than hex numbers, so it is ok to be slightly out. I guess in general we want to show the one that is right for the newer architecture.